rest_framework/views.py

```diff
@@ -350,9 +350,13 @@ def check_throttles(self, request):
         Check if request should be throttled.
         Raises an appropriate exception if the request is throttled.
         """
+        throttle_durations = []
         for throttle in self.get_throttles():
             if not throttle.allow_request(request, self):
-                self.throttled(request, throttle.wait())
+                throttle_durations.append(throttle.wait())
+
+        if throttle_durations:
+            self.throttled(request, max(throttle_durations))
```

Comment:
Looks good yup, but let's keep it seperate to the "include throttled requests in throttling limits"

Reply:
Which duration should we return then ? Now that we are checking them all we need to pick one.

Reply:
What I mean is that "Apply the throttling to all throttle clases" should be seperate to "Accumulate the throttling count, even for throttled requests".

Reply:
Ah, ok. I'll do that, it was just more annoying for tests in the second PR since they'd have to be adjusted once this PR lands :)

Reply:
@diox I'm happy to rebase if you include the first PR's commits in the second PR.